### PR TITLE
[RFR] Improve styleability, add new components

### PR DESCRIPTION
**Pull request by fzaninotto**
This PR is tre result of the efforts of implementing [admin-on-rest-demo](https://github.com/marmelab/admin-on-rest-demo) further.

* [x] Align datagrid first column to the page title
* [x] [BC Break] Rename `style` to `elStyle` and let `style` override container element
* [x] Fix warning for fields with no source and no label
* [x] Hide resources in the Menu when they don't have a list view
* [x] Fix FilterButton for fields without label
* [x] Make BooleanInput work without label
* [x] Add `<NumberField>`, `<AmountField>`, and `<FunctionField>`, with tests and doc

image
image
image

### CODE REVIEW

File: docs/Fields.md [hunk 1]
```diff
@@ -111,11 +124,35 @@ This component accepts a `showTime` attribute (false by default) to force the di
...
+## `<NumberField>`
+
+Displays a number, right aligned.
+
+The `decimals` prop allows to set the number of decimals (default to 0).
+
+```js
+import { NumberField }  from 'admin-on-rest/lib/mui';
+
+<AmountField source="total" decimals={2} />
```
Comment (djhi) [inline docs/Fields.md]: s/AmountField/NumberField/

File: src/mui/field/AmountField.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React, { PropTypes } from 'react';
+import get from 'lodash.get';
+
+/**
+ * @example
+ * <AmountField source="total" currency="€" decimals={2} />
+ */
+const AmountField = ({ record, source, currency, decimals, elStyle }) => record ?
+    <span style={elStyle}>{currency}{get(record, source).toFixed(decimals)}</span> :
```
Comment (djhi) [inline src/mui/field/AmountField.js]: Some currencies require their symbol to be put **after** the number (`€` for example)

Comment (djhi) [inline src/mui/field/AmountField.js]: This is a prb for other libraries such as [numeral.js](http://numeraljs.com/)

Comment (jpetitcolas) [inline src/mui/field/AmountField.js]: Agreed. I think this field should be user responsibility IMO. It induces too much complexity for a field which is quite specific.

Comment (djhi) [inline src/mui/field/AmountField.js]: Or we can add a simple `symbolPosition` prop with `left` by default and let the user deals with the auto detection if he needs it.

**Comment by fzaninotto** [comment]
Review applied

File: src/mui/field/NumberField.js [hunk 3]
```diff
@@ -1,25 +1,56 @@
 import React, { PropTypes } from 'react';
 import get from 'lodash.get';
 
+const hasNumberFormat = !!(typeof Intl === 'object' && Intl && typeof Intl.NumberFormat === 'function');
+
 /**
+ * Display a numeric value as a locale string.
+ *
+ * Uses Intl.NumberFormat() if available, passing the locales and options props as arguments.
```
Comment (djhi) [inline src/mui/field/NumberField.js]: :+1:

Comment (djhi) [inline src/mui/field/NumberField.js]: So this is the user responsibility to include an Intl polyfill if needed ?

**Pull request merged**